### PR TITLE
Openlineage sql

### PR DESCRIPTION
**Pull request by rxm7706**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Running conda-build locally, everything worked fine - but on conda-forge the build does not seem to work for linux without setting   error_overlinking: false.

### CODE REVIEW

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

**Comment by rxm7706** [comment]
@conda-forge-admin ready for review!

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

**Comment by rxm7706** [comment]
@conda-forge-admin, please restart ci

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

File: recipes/openlineage-sql/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/openlineage_sql-{{ version }}.tar.gz
+  sha256: 8e78db662edf1efe94da182ecfc98dbdc6ccf93a0489703c5d8a49d551760e67
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  error_overlinking: false
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:16]: I'm not a Rust reviewer but why is this needed here? Usually it hides over-linking problems that should be addressed. Use it with caution.

Comment (rxm7706) [reply recipes/openlineage-sql/meta.yaml:16]: Thanks @ocefpaf I will reach out to rust help - I dont have much experience with rust packages, and I didnt get any overlinking issues locally - not sure what I am missing here.

**Comment by rxm7706** [comment]
> > @conda-forge-admin, please rerender
> 
> There is no rerender in staged-recipes. Only in the feedstocks.

Understood - one additional question - for a maintainers / admins review - do I have to keep the branch up to date - (by using update branch / update with rebase)

**Comment by ocefpaf** [comment]
> - for a maintainers / admins review - do I have to keep the branch up to date - (by using update branch / update with rebase)

No need for that in staged-recipes Unless is was sitting here for a long time and our CI configuration is stale but the reviewer will let you know that.

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

**Comment by rxm7706** [comment]
@conda-forge/help-python could you please review.
All tests pass, but trying to keep the package as noarch required missing dso. 
I also see this sample - which doesnt use noarch # Note sample: https://github.com/conda-forge/y-py-feedstock/blob/main/recipe/meta.yaml

File: recipes/openlineage-sql/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/openlineage_sql-{{ version }}.tar.gz
+  sha256: df146cb536d6d9a336afe073e73982477ef0dd1220e27975687355034e7d2daa
+
+build:
+  noarch: python
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:13]: This is probably a mistake in grayskull. If there is a compilation step it cannot be `noarch`.

```suggestion
  skip: True  # [py<38]
```

File: recipes/openlineage-sql/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  missing_dso_whitelist:
+    - '*/libgcc_s.so.1'  # [linux]
+    - '*/librt.so.1'  # [linux]
+    - '*/libpthread.so.0'  # [linux]
+    - '*/libdl.so.2'  # [linux]
+    - '*/libc.so.6'  # [linux]
+    - '*/ld-linux-x86-64.so.2'  # [linux]
+  ignore_run_exports:
+    - libgcc-ng
+    - sysroot_linux-64
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:25]: Let's make this arch and see if we still need it.

```suggestion
```

File: recipes/openlineage-sql/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+requirements:
+  build:
+    - {{ compiler('rust') }}
+  host:
+    - python >=3.7
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:31]:
```suggestion
    - python
```

File: recipes/openlineage-sql/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - maturin >=1.0
+    - pip
+  run:
+    - python >=3.7
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:35]:
```suggestion
    - python
```

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

File: recipes/openlineage-sql/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  sha256: df146cb536d6d9a336afe073e73982477ef0dd1220e27975687355034e7d2daa
+
+build:
+  skip: True  # [py<38]
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('rust') }}
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:19]: Let's see if this fixes the warnings/errors we are getting.

```suggestion
    - {{ compiler('c') }}
    - {{ compiler('rust') }}
```

File: recipes/openlineage-sql/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/openlineage_sql-{{ version }}.tar.gz
+  sha256: df146cb536d6d9a336afe073e73982477ef0dd1220e27975687355034e7d2daa
+
+build:
+  skip: True  # [py<38]
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:14]:
```suggestion
  script:
    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
    - {{ PYTHON }} -m pip install . -vv
```

File: recipes/openlineage-sql/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
```
Comment (ocefpaf) [inline recipes/openlineage-sql/meta.yaml:20]:
```suggestion
    - {{ compiler('rust') }}
    - cargo-bundle-licenses
```

**Review by ocefpaf — COMMENTED** [review]
I think we need this to bundle the 3rd party licenses cargo may be pulling.
